pkg/cmd/experimental/registry/registry.go

```diff
@@ -0,0 +1,267 @@
+package registry
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	kclientcmd "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/cmd/util/variable"
+	configcmd "github.com/openshift/origin/pkg/config/cmd"
+	dapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/generate/app"
+)
+
+const longDesc = `
+Install or configure a Docker registry for OpenShift
+
+This command helps to setup a Docker registry integrated with OpenShift to provide
+notifications when images are pushed. With no arguments, the command will check for the existing registry
+service called 'docker-registry' and perform some diagnostics to ensure the registry is properly
+configured and functioning.
+
+If a registry service does not exist, the --create flag can be passed to
+create a deployment configuration and service that will run the registry. If you are
+running your registry in production, you should pass --replicas=2 or higher to ensure
+you have failover protection.
+
+Examples:
+  Check the default Docker registry ("docker-registry"):
+
+  $ %[1]s %[2]s
+
+  See what the registry would look like if created:
+
+  $ %[1]s %[2]s -o json
+
+  Create a registry if it does not exist with two replicas:
+
+  $ %[1]s %[2]s --create --replicas=2
+
+  Use a different registry image and see the registry configuration:
+
+  $ %[1]s %[2]s -o yaml --images=myrepo/docker-registry:mytag
+
+ALPHA: This command is currently being actively developed. It is intended to simplify
+  the tasks of setting up a Docker registry in a new installation.
+`
+
+type config struct {
+	Type          string
+	ImageTemplate variable.ImageTemplate
+	Ports         string
+	Replicas      int
+	Labels        string
+	Volume        string
+	HostMount     string
+	Create        bool
+	Credentials   string
+}
+
+const defaultLabel = "docker-registry=default"
+
+func NewCmdRegistry(f *clientcmd.Factory, parentName, name string, out io.Writer) *cobra.Command {
+	cfg := &config{
+		ImageTemplate: variable.NewDefaultImageTemplate(),
+
+		Labels:   defaultLabel,
+		Ports:    "5000",
+		Volume:   "/registry",
+		Replicas: 1,
+	}
+
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s", name),
+		Short: "Install and check OpenShift Docker registry",
+		Long:  fmt.Sprintf(longDesc, parentName, name),
+
+		Run: func(cmd *cobra.Command, args []string) {
+			var name string
+			switch len(args) {
```

Comment:
Generally, if you're dealing with multiple conditions (and we will eventually here) switch is just as reasonable as if.  In fact, Go usually prefers switch over else blocks.

----- Original Message -----
> > +		ImageTemplate: variable.NewDefaultImageTemplate(),
> > +
> > +		Labels:   defaultLabel,
> > +		Ports:    "5000",
> > +		Volume:   "/registry",
> > +		Replicas: 1,
> > +	}
> > +
> > +	cmd := &cobra.Command{
> > +		Use:   fmt.Sprintf("%s", name),
> > +		Short: "Install and check OpenShift Docker registry",
> > +		Long:  fmt.Sprintf(longDesc, parentName, name),
> > +
> > +		Run: func(cmd *cobra.Command, args []string) {
> > +			var name string
> > +			switch len(args) {
> 
> just want to understand, is there a reason (in Go) to prefer switch over if?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1100/files#r25151115